kasa/tests/test_sslaestransport.py

```diff
@@ -0,0 +1,397 @@
+    assert "result" in res
+    assert (
+        "Received unencrypted response over secure passthrough from 127.0.0.1"
+        in caplog.text
+    )
+
+
+async def test_port_override():
+    """Test that port override sets the app_url."""
+    host = "127.0.0.1"
+    port_override = 12345
+    config = DeviceConfig(
+        host, credentials=Credentials("foo", "bar"), port_override=port_override
+    )
+    transport = SslAesTransport(config=config)
+
+    assert str(transport._app_url) == f"https://127.0.0.1:{port_override}"
+
+
+class MockSslAesDevice:
+    BAD_USER_RESP = {
+        "error_code": SmartErrorCode.SESSION_EXPIRED.value,
+        "result": {
+            "data": {
+                "code": -60502,
+            }
+        },
+    }
+
+    BAD_PWD_RESP = {
+        "error_code": SmartErrorCode.INVALID_NONCE.value,
+        "result": {
+            "data": {
+                "code": SmartErrorCode.SESSION_EXPIRED.value,
+                "encrypt_type": ["3"],
+                "key": "Someb64keyWithUnknownPurpose",
+                "nonce": "1234567890ABCDEF",  # Whatever the original nonce was
+                "device_confirm": "",
+            }
+        },
+    }
+
+    class _mock_response:
+        def __init__(self, status, request: dict):
+            self.status = status
+            self._json = request
+
+        async def __aenter__(self):
+            return self
+
+        async def __aexit__(self, exc_t, exc_v, exc_tb):
+            pass
+
+        async def read(self):
+            if isinstance(self._json, dict):
+                return json_dumps(self._json).encode()
+            return self._json
+
+    def __init__(
+        self,
+        host,
+        *,
+        status_code=200,
+        want_default_username: bool = False,
+        do_not_encrypt_response=False,
+        send_response=None,
+        sequential_request_delay=0,
+        send_error_code=0,
+        secure_passthrough_error_code=0,
+        digest_password_fail=False,
+    ):
+        self.host = host
+        self.status_code = status_code
+        self.send_error_code = send_error_code
+        self.secure_passthrough_error_code = secure_passthrough_error_code
+        self.do_not_encrypt_response = do_not_encrypt_response
+        self.send_response = send_response
+        self.http_client = HttpClient(DeviceConfig(self.host))
+        self.inner_call_count = 0
+        self.token = "".join(random.choices(string.ascii_uppercase, k=32))  # noqa: S311
+        self.sequential_request_delay = sequential_request_delay
+        self.last_request_time: float | None = None
+        self.sequential_error_raised = False
+        self.handshake1_complete = False
+        self.server_nonce = secrets.token_bytes(8).hex().upper()
+        self.want_default_username = want_default_username
+        self.encryption_session: AesEncyptionSession | None = None
+        self.digest_password_fail = digest_password_fail
```

Comment:
Most code lines in this class are rather tightly w/o whitespaces or grouping, I'd suggest making it a bit nicer to read by adding some new lines in relevant places & grouping relevant items together.

Reply:
Cleaned this up and removed a lot of attributes not currently used by these tests.